File: ruby/zenn_14.rb
# 次の変数engineersに格納された
# エンジニアの種類(フロントエンド、バックエンド)の数を種類を集計し、
# 一つのオブジェクトに格納して下さい。

def engType

  engineers = {
    a: "Frontend",
    b: "Backend",
    c: "Backend",
    d: "Frontend",
    e: "Frontend",
    f: "Backend",
    g: "Backend"
  }

  sum = {}
  sum['Frontend'] = engineers.select{|k, v| v == "Frontend"}.size
  sum['Backend'] = engineers.select{|k, v| v == "Backend"}.size
  p sum

end

engType

File: js/zenn_14.js
// 次の変数engineersに格納されたエンジニアの種類(フロントエンド、バックエンド)の数を
// 種類を集計し、
// 一つのオブジェクトに格納して下さい。

const engineers = [
  { type: "Frontend" },
  { type: "Backend" },
  { type: "Backend" },
  { type: "Frontend" },
  { type: "Frontend" },
  { type: "Backend" },
  { type: "Backend" }
];

const engType = engineers.reduce(
  (sum, { type }) => {
    if (type === "Frontend") {
      sum.Frontend++;
    } else {
      sum.Backend++;
    }
    return sum;
  },
  { Frontend: 0, Backend: 0 }
);
console.log(engType);//{Frontend: 3, Backend: 4}
